File: vue-app/src/shared/classes/ProtelUser.ts
import type { IProtelUser } from '../interfaces/IProtelUser'

export class ProtelUser implements IProtelUser {
  id: string | undefined
  createdAt: Date
  updatedAt: Date
  email: string
  allowedCampIDs: string

  constructor() {
    this.id = undefined
    this.createdAt = new Date()
    this.updatedAt = new Date()
    this.email = ''
    this.allowedCampIDs = ''
  }
}

File: backend-middleware/src/main/java/tpms/backend_middleware/models/ProtelUser.java
package tpms.backend_middleware.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import tpms.backend_middleware.interfaces.IModel;
import lombok.Data;

@Data
@Document
public class ProtelUser implements IModel {
    @Id
    private String id;
    private Date createdAt = new Date();
    private Date updatedAt = new Date();
    private String email = "";
    private String allowedCampIDs = "";
}
